src/args.js

```diff
@@ -0,0 +1,79 @@
+'use strict'
+const _ = require('redash')
+const min = require('minimist')
+const pkg = require('../package.json')
+
+/* istanbul ignore next */
+const processArgs = process.argv[0] === 'node' ? 1 : 2
+
+const args = {
+  /**
+   * @property {object} Arguments passed to the instance
+   */
+  raw: min(process.argv.slice(processArgs)),
+  /**
+   * @property {object} Available run-time flags
+   * - @property {string} action - the method to call when parsing
+   * - @property {string} prop - the prop to set when parsing
+   * - @property {string} help - the help text to display to the user
+   */
+  available: {
+    h: { action: 'showHelp', help: 'Displays help and usage' },
+    v: { action: 'showVersion', help: 'Displays the current installed version' },
+    e: { prop: 'exec', help: 'Run a custom command instead of defined task' },
+    f: { prop: 'from', help: 'Run with specified docker image' },
+    c: { prop: 'configPath', help: 'Run with custom config file path' }
+  },
+  /**
+   * Displays the help and usage message
+   */
+  showHelp: () => {
+    let help = ''
+    help += `  ${pkg.name} v.${pkg.version}\n`
+    help += `  Usage: [${_.keys(pkg.bin).join('|')}] task [options]\n`
+    help += _.keys(args.available).reduce((p, c) => `${p}\n    -${c}  ${args.available[c].help}`, '')
```

Comment:
Whenever I find myself having to access both an objects keys and values `toPairs` is often a better fit than just `keys`, since it eliminates the need to look that key up.

```js
help += _.toPairs(args.available).reduce((p, [key, value]) => `${p}\n    -${key} ${value.help}`, '')
```

or...

```js
help += _.pipe(_.toPairs
               _.map(([k, v]) => `  -${k} ${v.help}`)
               _.join('\n'))(args.available)
```

Generally, the more you use functions that help you do what you're trying to do, the more straightforward code you'll write; or at a minimum, it well have you more closely focus on the discrete steps involved.